internal/controllers/common/resource_definitions/resource_definitions.go

```diff
@@ -653,30 +653,61 @@ func NewJfrDatasourceContainer(cr *operatorv1beta1.Cryostat, imageTag string) co
 	}
 }
 
-func NewExporterService(cr *operatorv1beta1.Cryostat) *corev1.Service {
+func NewCoreService(cr *operatorv1beta1.Cryostat) *corev1.Service {
+	// Initialize with defaults
+	svcType := corev1.ServiceTypeClusterIP
+	httpPort := int32(8181)
+	jmxPort := int32(9091)
+	labels := map[string]string{}
+	annotations := map[string]string{}
+
+	// Override with any user-specified properties
+	if cr.Spec.ServiceOptions != nil && cr.Spec.ServiceOptions.CoreConfig != nil {
+		config := cr.Spec.ServiceOptions.CoreConfig
+		if config != nil {
+			if config.ServiceType != nil {
+				svcType = *config.ServiceType
+			}
+			if config.HTTPPort != nil {
+				httpPort = *config.HTTPPort
+			}
+			if config.JMXPort != nil {
+				jmxPort = *config.JMXPort
+			}
+			if config.Labels != nil {
+				labels = config.Labels
+			}
+			if config.Annotations != nil {
+				annotations = config.Annotations
+			}
+		}
+	}
+
+	// Add required labels, overriding any user-specified labels with the same keys
+	labels["app"] = cr.Name
```

Comment:
Would it make sense to emit a warning or something if the user has specified labels that we are going to override?

Reply:
I'm not sure how best to go about this. This method gets called each iteration of the reconcile loop, regardless of whether it ends up creating or changing the service. With either an Event or log message, it could get noisy. It's at least documented in the property description.

Reply:
Ah, yea that's true.